momepy/elements.py

```diff
@@ -668,30 +655,23 @@ def __init__(self, tessellation, edges, buildings, id_name, unique_id, **kwargs)
                 cells_copy.dissolve(by="component").explode().reset_index(drop=True)
             )
         blocks[id_name] = range(len(blocks))
-        blocks[blocks.geometry.name] = gpd.GeoSeries(
-            pygeos.polygons(blocks.exterior.values.data), crs=blocks.crs
-        )
```

Comment:
I originally did this to cleanup interior after complex dissolve but it seems that GEOS got much better on it so it may not be needed. This is what was causing the issue in the first place, so if we want some cleaning procedure, it needs to be different.